ocamltest/ocaml_actions.ml

```diff
@@ -1042,7 +1089,9 @@ let run_test_program_in_toplevel (toplevel : Ocaml_toplevels.toplevel) log env =
           [
             toplevel_name;
             Ocaml_flags.toplevel_default_flags;
-            toplevel#flags;
+            (match Ocaml_toplevels.backend toplevel with
+             | Ocaml_backends.Native -> "-S"
+             | _ -> "");
```

Comment:
There is a small breach of abstraction level here again. If the dependency structure allows, I would be in favor of just including `Ocaml_flags.toplevel_flags toplevel`, which would include both the default flags and this one. (`Ocaml_flags` provides `_default_` choices for many categories of flags, but it also provide some similar full-flags functions like `runtime_flags`.)